src/index.ts

```diff
@@ -146,15 +115,11 @@ export const createRouter = <
     replace?: boolean | undefined;
     target?: React.HTMLAttributeAnchorTarget | undefined;
   }) => {
-    const active = useSubscription(
-      React.useMemo(
-        () => ({
-          getCurrentValue: () => href === currentLocation.url,
-          subscribe,
-        }),
-        [href],
-      ),
-    );
+    const location = useLocation();
```

Comment:
This will force a re-render of the component using the hook at each location update, not only when `active` boolean value change.